spec.bs

```diff
@@ -651,6 +677,169 @@ traversables</a> section.
   1. Return |navigables|.
 </div>
 
+<h3 id=focusing-changes>Modifications to the focusing algorithms</h3>
+
+*This section is non-normative*.
+
+The [[HTML]] standard defines how to handle focusing elements and {{Window}}s, both by user gesture
+and through script-initiated APIs. Since fenced frames are designed to prevent communication across
+a fenced frame boundary, we need to handle focusing carefully. This is because when focus is pulled
+across a fenced frame boundary, contexts on both sides of the boundary can detect that change, which
+can be used to open a communication channel between a <{fencedframe}> and its embedder.
+
+We do this is by not allowing the [=focusing steps=] to move script-initiated focus across a fenced
+frame boundary.
+
+<p class=example id=user-initiated-focus>When a user clicks on an element like a <{button}> inside a
+<{fencedframe}> while another element *outside* of the <{fencedframe}> is [=focused=], because this
+is a user-initiated action, the [=focusing steps=] will allow the <{button}> to [=gain focus=].
+Without allowing that, no element inside of a <{fencedframe}> would never be able [=gain
+focus=].</p>
+
+<p class=example id=script-initiated-focus>If we were to continue blindly allowing all elements to
+be [=focused=] via the {{HTMLOrSVGElement/focus()}} method as is the status quo before this
+specification, a [=fenced navigable container=] and its [=fenced navigable container/fenced
+navigable=] could use a sequence of {{HTMLOrSVGElement/focus()}} calls to send arbitrary data across
+the fenced frame boundary, which is a privacy leak. To avoid this, we effectively "fence" the
+{{HTMLOrSVGElement/focus()}} method, which sacrifices some functionality for privacy.</p>
+
+<h4 id=focusing-steps-changes>Changes to the [=focusing steps=] algorithm</h4>
+
+<div algorithm=focusing-steps-patch>
+  Modify the [=focusing steps=] to take a new optional [=boolean=] argument <dfn
+  lt="focus-unfenced">unfenced</dfn> that defaults to false.
+
+  Add a new step after step 3 of the algorithm (that changes new focus target) that reads:
+
+  4. If |new focus target| is a [=fenced navigable container=] with non-null
+     [=fenced navigable container/fenced navigable=], then set <var>new focus target</var> to the
+     [=fenced navigable container/fenced navigable=]'s [=navigable/active document=].
+
+  Add a new step after step 8 (step 7 in the existing version) of the algorithm (that defines |old
+  chain| and |new chain|) that reads:
+
+  9. If [=focus-unfenced|unfenced=] is false, |new chain| [=list/contains=] an element |element|
+     whose [=node navigable=]'s [=navigable/traversable navigable=] is a [=fenced navigable
+     container/fenced navigable=], and |old chain| does not also [=list/contain=] |element|, then
+     return.
+
+     Note: This is how we bail-out early just before calling the [=focus update steps=], in the case
```

Comment:
I added this `Note:`. Does it capture the intent correctly? If so (that is, if I understand what our aim is here), then I think the above step is not adequate. If you move focus from one element inside of a fenced frame to another element inside of a fenced frame, I think the logic in your step will trigger and then we'll mistakenly early-return, right? Does that logic make sense?

Reply:
I think it should be checking for a document instead of an element. That should correctly handle the case of switching focus between 2 elements in a fenced frame.

Both focus chains will contain the fenced frame root's document at that point, and the function will not early return. If we switch focus from one fenced frame to another, now both of the fenced frame root documents will only exist in one of the focus chains, and the function will early return.